numpy/linalg/setup.py

```diff
@@ -53,8 +53,41 @@ def get_lapack_lite_sources(ext, build_dir):
         extra_info=lapack_info,
         libraries=['npymath'],
     )
+
+    # TODO: link extension robustly on non-Linux
+    if len(get_info('openblas')) > 2:
+        # if openblas is being used to compile NumPy
+        # build an extension to gather more information
+        # about the openblas version used
+        config.add_extension('openblas_config',
+                             sources=['openblas_config.c'],
+                             extra_info=lapack_info,
+                             libraries=['openblas'],
+                             )
     return config
 
 if __name__ == '__main__':
     from numpy.distutils.core import setup
     setup(configuration=configuration)
+    import numpy.distutils.system_info
```

Comment:
this block never gets executed -- I need to move this code somewhere more appropriate; perhaps at some point of linalg initialization instead of in the distutils spaghetti

Reply:
I suggest to remove that code block then, doesn't really seem useful.

Reply:
And I'd prefer to have the mess contained in `numpy.distutils`. Also because then it's available to `scipy.linalg` and other users.